rosbag2_cpp/package.xml

```diff
@@ -27,6 +27,8 @@
   <depend>shared_queues_vendor</depend>
 
   <exec_depend>rosbag2_storage_default_plugins</exec_depend>
+  <!-- TODO: remove dependency when rosbag2_storage_default_plugins depends on MCAP -->
+  <exec_depend>rosbag2_storage_mcap</exec_depend>
```

Comment:
It seems reasonable to me to add `rosbag2_storage_mcap` to `rosbag2_storage_default_plugins` dependency as part of this PR. What that does in effect is start shipping the plugin as part of `rosbag2` metapackage, without changing the default. This dependency for `rosbag2_cpp` does that anyways, but in an awkward spot (this should be a `test_depend` if it were to be here though, I don't know why default_plugins is an exec_depend. I think we've had this conversation. Maybe change it now?)

Reply:
would this cause `rosbag2_storage_mcap` to ship with rosbag2 if it were `test_depend` rather than exec_depend?

Reply:
Test depends don't get built into the packaging for release